### ISSUE TITLE
FileInputStream constructor should throw FileNotFoundException on invalid file names

### ISSUE THREAD

**Issue by mccartney** [scala-native/scala-native#765]
A difference in behavior between JVM and Scala-Native (follows #763)

# Steps to reproduce

Try to open a file which doesn't exist:
```
  		val fis = new java.io.FileInputStream("/this/path/has/no/chance/to/exist")
  		println(fis.read)
  		fis.close
```

# Observed in Scala-Native
```java.io.IOException: couldn't read from the file```
happening during the first read

# Expected (JVM behavior):
```java.io.FileNotFoundException```
happening already on the construction call
as specified by [FileInputStream javadoc](https://docs.oracle.com/javase/8/docs/api/java/io/FileInputStream.html#FileInputStream-java.lang.String-)

**Comment by mccartney** [comment]
Moreover, it seems the JVM behavior is to throw exception already on creating a new stream: `new java.io.FileInputStream("/this/path/has/no/chance/to/exist")` without even reading from it.  Scala-Native fails only on reading.

**Comment by Duhemm** [comment]
Fixed in #769 / #835
